docs/DeveloperGuide.md

```diff
@@ -1,34 +1,956 @@
+6. Click `Open as Project`.
+7. Click `OK` to accept the default settings.
+
+### 2.3. Verifying the setup
+
+1. Run the `seedu.duke.Duke`.
+2. Try a few commands.
+3. [Run the tests](#61-running-tests) to ensure they all pass.
+
+### 2.4. Configure the coding style
+
+If using IDEA, follow the guide [[se-edu/guides] IDEA: Configuring the code style](https://se-education.org/guides/tutorials/intellijCodeStyle.html)
+to set up IDEA’s coding style to match ours.
+
+>Optionally, you can follow the guide [[se-edu/guides] Using Checkstyle](https://se-education.org/guides/tutorials/checkstyle.html)
+>to find how to use the CheckStyle within IDEA e.g., to report problems as you write code.
+
+## 3. Design
+
+The following section describes the design and implementation of the product. UML diagrams and code snippets are used
+to explain some aspects of the code. If you are unfamiliar with UML, the diagrams should still be fairly
+understandable. However, you may wish to consult [[CS2113/T] Modeling](https://nus-cs2113-ay2021s1.github.io/website/se-book-adapted/chapters/modeling.html) for a quick introduction to UML.
+
+### 3.1 Architecture (Neil)
+
+**How the architecture components interact with each other**
+The Sequence Diagram below shows how the components interact with each other for the scenario...
+
+/* work in progress */
+
+### 3.2 UserInterface Component (Neil)
+
+The UserInterface Component is made up of `AppMode`, `AppState`, `CliMessages`, `CliUserInterface`, `InputParser`.
+
+The `UserInterface` component,
+
+* Executes user commands using the `Commands` component.
+* // how it interacts with the other components
+
+/* TODO explain the various variables and methods */
+
+### 3.3. Commands Component (Neil)
+
+![UML diagram for Timetable Commands](/diagrams/class/jpeg/timetable_commands.jpg)
+
+![UML diagrams for Notebook Commands](/diagrams/class/jpeg/notebook_commands.jpg)
+
+/* to insert UML diagram */
+
+The `Commands` component,
+
+// TODO add components
+
+### 3.4. Tasks Component
+
+/* to insert UML diagram */
+
+### 3.5 Notebooks Component
+
+![UML diagram for Notebooks](/diagrams/class/jpeg/notebooks.jpg)
+
+/* TODO explain the various variables and methods */
+
+### 3.6. Storage Component
+
+/* to insert UML diagram */
+
+The `Storage` component,
+
+* can save the name of user in a .txt file and read it back.
+
+* can save the Tasklist and NotebookShelf data in a .txt file and read it back.
+
+## 4. Implementation
+
+The following section describes the implementation of certain key features in the current version of Zer0Note. It also
+provides some background into our (the original developers of Zer0Note) thinking and the rationale behind the
+decisions.
+
+### 4.1. Mode Switch Feature
+
+#### 4.1.1. Implementation
+
+/* work in progress */
+
+#### 4.1.2. Design Considerations
+
+/* work in progress */
+
+### 4.2. Timetable Mode
+
+#### 4.2.1. TaskList Management Feature
+
+##### 4.2.1.1. Implementation
```

Comment:
Perhaps you could make this heading a bit larger? I find it a bit strange if the size of the heading text is smaller than the body text below it.

Reply:
We've been having some issues with headings; we're still experimenting with different heading levels to make it the most readable. Thanks for the feedback, we'll try to fix this as best as possible.